boxoffice/models/line_item.py

```diff
@@ -26,21 +26,23 @@ def make_ntuple(item_id, base_amount, **kwargs):
 
 class Assignee(BaseMixin, db.Model):
     __tablename__ = 'assignee'
+    __table_args__ = (db.UniqueConstraint('line_item_id', 'current'),)
 
     # lastuser id
     user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=True)
     user = db.relationship('User', backref=db.backref('assignees', cascade='all, delete-orphan'))
 
+    line_item_id = db.Column(None, db.ForeignKey('line_item.id'), nullable=False)
+    line_item = db.relationship('LineItem',
+        backref=db.backref('assignees', cascade='all, delete-orphan', lazy='dynamic'))
+
     fullname = db.Column(db.Unicode(80), nullable=False)
     #: Unvalidated email address
     email = db.Column(db.Unicode(254), nullable=False)
     #: Unvalidated phone number
     phone = db.Column(db.Unicode(16), nullable=True)
     details = db.Column(JsonDict, nullable=False, default={})
-
-    # Track the assignee from whom the line_item was transferred from
-    previous_id = db.Column(None, db.ForeignKey('assignee.id'), nullable=True)
-    previous = db.relationship('Assignee', uselist=False)
+    current = db.Column(db.Boolean, default=False, nullable=True)
```

Comment:
No default here. `False` is an invalid value. (The default for `default` is `None`, which is what you want.)